### PR TITLE
Add extra function to pay rewards to external address

### PR DESCRIPTION
**Pull request by brianmcmichael**
Overloads the bark function so that a keeper can send the rewards to a different address.

### CODE REVIEW

File: src/dog.sol [hunk 1]
```diff
@@ -202,11 +202,15 @@ contract Dog {
             });
         }
 
-        vat.suck(address(vow), msg.sender, add(milk.tip, wmul(due, milk.chip)));
+        vat.suck(address(vow), guy, add(milk.tip, wmul(due, milk.chip)));
 
         emit Bark(ilk, urn, dink, dart, due, milk.clip, id);
     }
 
+    function bark(bytes32 ilk, address urn) external returns (uint256 id) {
```
Comment (gbalabasquer) [inline src/dog.sol:210]: I'd tend to prefer to not have this function directly if we want to have a different address for the `tip`.
So only keep `bark(bytes32 ilk, address urn, address guy)` and make it `external`.

File: src/dog.sol [hunk 2]
```diff
@@ -150,7 +150,7 @@ contract Dog {
     function chop(bytes32 ilk) external view returns (uint256) { return ilks[ilk].chop; }
 
     // --- CDP Liquidation: all bark and no bite ---
-    function bark(bytes32 ilk, address urn) external returns (uint256 id) {
+    function bark(bytes32 ilk, address urn, address guy) public returns (uint256 id) {
```
Comment (gbalabasquer) [inline src/dog.sol:153]: I think we should use `usr` which is the standard now.

Comment (brianmcmichael) [reply src/dog.sol]: fixed

File: src/dog.sol (hunk 1 above)
Comment (brianmcmichael) [reply src/dog.sol]: removed the overload

File: src/dog.sol (hunk 2 above)
Comment (gbalabasquer) [reply src/dog.sol]: I think you missed to push the change or something is not getting updated.

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Review by godsflaw — APPROVED** [review]
LGTM, remember to add this to the redo() when we add incentives there too.

**Pull request merged**